app/services/mapping/goodsshipment/WarehouseBuilder.scala

```diff
@@ -25,20 +25,28 @@ object WarehouseBuilder {
   def build(implicit cacheMap: CacheMap): Warehouse =
     cacheMap
       .getEntry[WarehouseIdentification](WarehouseIdentification.formId)
+      .filter(isDefined)
       .map(createWarehouse)
       .orNull
 
+  private def isDefined(warehouse: WarehouseIdentification): Boolean =
+    !warehouse.identificationNumber.getOrElse("").isEmpty
+
   private def createWarehouse(data: WarehouseIdentification): Warehouse = {
+    val warehouse = new Warehouse()
 
-    val id = new WarehouseIdentificationIDType()
-    id.setValue(data.identificationNumber.map(_.drop(1).toString).getOrElse(""))
+    if (!data.identificationNumber.map(_.drop(1).toString).getOrElse("").isEmpty) {
+      val id = new WarehouseIdentificationIDType()
+      id.setValue(data.identificationNumber.map(_.drop(1).toString).getOrElse(""))
```

Comment:
`data.identificationNumber.map(_.drop(1).toString).getOrElse("")`

We can save it in variable and declare before if. Reusing in condition and here.

The same below

Reply:
I have removed that condition. I am using `filter`